### ISSUE TITLE
Source DatoCMS Single Instance

### ISSUE THREAD

**Issue by thetre97** [gridsome/gridsome#541]
## Description

Single-instance DatoCMS models appear in the playground, but return `null` when queried. For instance:

```graphql
datoCmsHome {
    introText
  }
```
returns
```json
"datoCmsHome": null,
```

However querying -
```graphql
allDatoCmsHome {
    edges {
      node {
        id
        introText
      }
    }
  }
```
*-* will return 
```json
"allDatoCmsHome": {
      "edges": [
        {
          "node": {
            "id": "962827",
            "introText": "I am a creative illustrator and graphic designer with more than 10 years of experience.\n\nOriginally from Toronto, currently based in London."
          }
        }
      ]
    ,
```

### Steps to reproduce

https://github.com/thetre97/gridsome-starter-datocms

Simply create a single instance DatoCMS record, and query with the plugin.

### Expected result

`datoCmsHome { ...` should return the single instance data.  

### Actual result

```json
"datoCmsHome": null
```

### Environment

```json
"dependencies": {
    "@gridsome/source-datocms": "^0.1.0",
    "gridsome": "^0.6.6"
  },
```

**Comment by gilesbutler** [comment]
This is still an issue in `0.7.7` too.

A workaround is to query the instance by it's id...
```
datoPageContent(id: "1543729") {
    title
  }
```
This is pretty brittle though as it'll break if the id changes. 

I'm not sure if it's related but you can't query by `path` either anymore...
```
datoPageContent(path: "/home") {
    heroTitle
  }
```
This doesn't work anymore.

**Comment by matjack1** [comment]
Hello @gilesbutler @thetre97 sorry for this question but I'm not familiar with Gridsome, while I can help on DatoCMS side :)

From reading the docs ( https://gridsome.org/docs/collections/#collections-in-graphql ) looks like the singular `datoCmsHome` should be queried by id, isn't it?

It makes sense that since it's only one you should just get that, but how can I tell that to Gridsome when using `collection.addNode` ( https://gridsome.org/docs/data-store-api/#collectionaddnodeoptions )? Is there an option for that, where can I find that?

Happy to help more here if I can! :)

**Comment by gilesbutler** [comment]
Hey @matjack1,

Yes that's correct, singular items should be queried by id for now.

@hjvedvik clarified this in the Discord chat last week...
> The id is probably the only option for now, unfortunately.. Gridsome doesn't modify it so it should be persistent when coming from the DatoCMS plugin. But we are going to require unique IDs for every node in the next version og Gridsome, so it might change when v0.8 is published. We are also planning to add all the filter inputs to single node fields in v0.8 and that will let you get a node by any field.

As for your second question, I'm not sure I fully understand but isn't what you're referring to already happening in the source-datocms plugin on [line 105](https://github.com/gridsome/gridsome/blob/a4e22d6e8f21be360a7e750dcfea27180910cbe4/packages/source-datocms/index.js#L105)?

My only concern with querying by id is how would that work across environments. Say you have a Dato dev environment and a prod environment too, wouldn't that id be different between them?

**Comment by hjvedvik** [comment]
@matjack1 We can probably make Gridsome just return the first node in the collection if no id argument is provided :) I see that the DatoCMS GraphQL API is doing it.

And we have just implemented filters for single node fields. Expect it to be included in v0.8 :)

**Comment by gilesbutler** [comment]
Sounds like a great solution to me, thanks @hjvedvik

**Comment by matjack1** [comment]
yes @hjvedvik and @gilesbutler is it something that I can do in the plugin or it's something that you'll do gridsome-side?

**Comment by gilesbutler** [comment]
Looks like it's Gridsome side @matjack1.

I think @hjvedvik was saying it will be added to v0.8 if I read that right...

> Expect it to be included in v0.8 :)
